File: Data-Structures/Arrays/Exercises/Contains-Duplicate/Solution-1/solution1.py
import unittest

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        numsSize = len(nums)
        for i in range(numsSize):
            if i < numsSize - 1:
                for j in range(i+1, numsSize):
                    if nums[i] == nums[j]:
                        return True
        return False

class TestTwoSum(unittest.TestCase):
    def test_two_sum_1(self):
        self.assertEqual(Solution().containsDuplicate([1,2,3,1]), True)
    
    def test_two_sum_2(self):
        self.assertEqual(Solution().containsDuplicate([1,2,3,4]), False)
    
    def test_two_sum_3(self):
        self.assertEqual(Solution().containsDuplicate([1,1,1,3,3,4,3,2,4,2]), True)

if __name__ == '__main__':
    unittest.main()

File: Data-Structures/Arrays/Exercises/Shuffle-The-Array/Solution-1-Simple/solution1.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> solution;

        for (int i = 0; i < n; i++) {
            solution.push_back(nums[i]);
            solution.push_back(nums[n + i]);
        }

        return solution;
    }
};


int main() {
        vector<int> nums = {2,5,1,3,4,7};
        int n = 3;

        vector<int> solution = Solution().shuffle(nums, n);
        for (int i = 0; i < 2*n; i++)
            printf("%d ", solution[i]);
        printf("\n");

    return 0;
}